src/app/api/dashboard_service/dashboard.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { Http, Response } from '@angular/http';
+
+import { UserClass } from '../Classes/user';
+import { WeatherClass } from '../Classes/weather';
+import { LoginService } from '../login_service/login.service';
+import { HeaderService } from '../header_service/header.service';
+
+@Injectable()
+export class DashboardService {
+
+  constructor(private login: LoginService, private header: HeaderService, private http: Http) {
+  }
+
+  addToHistory(location: string) {
+
+    const current = this.login.getCurrentUser();
+    if (current.history.indexOf(location) === -1) {
+      current.history.push(location);
+      localStorage.setItem('currentUser', JSON.stringify(current));
+    }
+
+    const users = JSON.parse(localStorage.getItem('users')) || [];
+    for (let user of users) {
+      if (user.email === current.email && user.password === current.password) {
+        if (user.history.indexOf(location) === -1) {
+          user.history.push(location);
+        }
+        break;
+      }
```

Comment:
- [ ] get used to using array functions (safety and readability):
```
users.forEach((user) => {
       if (user.email === current.email && user.password === current.password) {
         if (user.history.some(historyLocation => historyLocation === location)) {
             user.history.push(location);
         }
       break;
       }
});
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
Nice, I didn't know for 'some' function. :)